packages/at_secondary_server/lib/src/verb/handler/keys_verb_handler.dart

```diff
@@ -0,0 +1,224 @@
+import 'dart:collection';
+import 'dart:convert';
+
+import 'package:at_commons/at_commons.dart';
+import 'package:at_persistence_secondary_server/at_persistence_secondary_server.dart';
+import 'package:at_secondary/src/connection/inbound/inbound_connection_metadata.dart';
+import 'package:at_secondary/src/constants/enroll_constants.dart';
+import 'package:at_secondary/src/enroll/enroll_datastore_value.dart';
+import 'package:at_secondary/src/server/at_secondary_impl.dart';
+import 'package:at_secondary/src/verb/handler/abstract_verb_handler.dart';
+import 'package:at_server_spec/at_server_spec.dart';
+import 'package:at_server_spec/at_verb_spec.dart';
+
+class KeysVerbHandler extends AbstractVerbHandler {
+  static Keys keys = Keys();
+
+  KeysVerbHandler(SecondaryKeyStore keyStore) : super(keyStore);
+
+  @override
+  bool accept(String command) => command.startsWith('keys:');
+
+  @override
+  Verb getVerb() {
+    return keys;
+  }
+
+  @override
+  Future<void> processVerb(
+    Response response,
+    HashMap<String, String?> verbParams,
+    InboundConnection atConnection,
+  ) async {
+    final keyVisibility = verbParams[visibility];
+    final atSign = AtSecondaryServerImpl.getInstance().currentAtSign;
+    bool hasManageAccess = false;
+    var connectionMetadata =
+        atConnection.getMetaData() as InboundConnectionMetadata;
+    final enrollIdFromMetadata = connectionMetadata.enrollmentId;
+    if (enrollIdFromMetadata == null) {
+      throw AtEnrollmentException(
+          'Keys verb cannot be accessed without an enrollmentId');
+    }
+    logger.finer('enrollIdFromMetadata:$enrollIdFromMetadata');
+    final key =
+        '$enrollIdFromMetadata.$newEnrollmentKeyPattern.$enrollManageNamespace';
+
+    var enrollData = await _getEnrollData(key, atSign);
+    if (enrollData != null) {
+      final enrollDataStoreValue =
+          EnrollDataStoreValue.fromJson(jsonDecode(enrollData.data!));
+      if (enrollDataStoreValue.approval?.state != 'approved') {
+        throw AtEnrollmentException(
```

Comment:
I don't see a unit test for this case ... if there isn't one, please can you add one?

Reply:
Has this been done?

Reply:
yes Gary.in keys_verb_test.dart unit test - "keys verb on enrollment which is not approved"